pkg/streamingpromql/types/hpoint_ring_buffer.go

```diff
@@ -134,7 +160,7 @@ func (b *HPointRingBuffer) Append(p promql.HPoint) error {
 
 		newSlice, err := b.pool.GetHPointSlice(newSize)
 		if err != nil {
-			return err
+			return &promql.HPoint{}, err
```

Comment:
We can just return `nil, err` here, there's no need to allocate an empty `HPoint`.